pallets/executors/src/lib.rs

```diff
@@ -1623,18 +1656,230 @@ pub mod pallet {
                 .collect()
         }
 
-        /// Execute a side effect via XBI
-        /// `xbi` could change to side effects and do the conversions into Xbi messages
-        ///
-        /// @maciejbaj look here TODO: you'll need to do some conversions from side effect into local instruction
-        fn execute_xbi(
-            origin: &T::Origin,
-            xbi: &mut xp_format::XbiFormat,
-        ) -> Result<
-            HandlerInfo<frame_support::weights::Weight>,
-            frame_support::dispatch::DispatchErrorWithPostInfo,
-        > {
-            T::InstructionHandler::handle(origin, xbi)
+        pub fn execute_side_effects_with_xbi(
+            origin: OriginFor<T>, // Active relayer
+            xtx_id: XExecSignalId<T>,
+            side_effect: SideEffect<T::AccountId, BalanceOf<T>>,
+            max_exec_cost: u128,
+            max_notifications_cost: u128,
+        ) -> DispatchResultWithPostInfo {
+            let executor = ensure_signed(origin.clone())?;
+            let state = <CandidateInfo<T>>::get(&executor).ok_or(Error::<T>::NoSuchCandidate)?;
```

Comment:
Don't think that check is necessary - executor as per latest design could be any valid account on t3rn.

Reply:
Removed